README.md
![alt XOOPS CMS](http://xoops.org/images/logoXoops4GithubRepository.png)
## xDonations for [XOOPS CMS 2.5.8+](https://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](LICENSE)
[![Scrutinizer Code Quality](https://img.shields.io/scrutinizer/g/mambax7/xdonations.svg?style=flat)](https://scrutinizer-ci.com/g/mambax7/xdonations/?branch=master)
[![Codacy Badge](https://api.codacy.com/project/badge/grade/2d27c0023ee54f0b9ba2b5d17a68b2a5)](https://www.codacy.com/app/mambax7/xdonations)
[![Code Climate](https://img.shields.io/codeclimate/github/mambax7/xdonations.svg?style=flat)](https://codeclimate.com/github/mambax7/xdonations)
[![SensioLabsInsight](https://insight.sensiolabs.com/projects/92d6efe7-3806-48c1-a47e-39e2130145b0/mini.png)](https://insight.sensiolabs.com/projects/92d6efe7-3806-48c1-a47e-39e2130145b0)
[![Latest Pre-Resease](https://img.shields.io/github/tag/XoopsModules25x/xdonations.svg?style=flat)](https://github.com/XoopsModules25x/xdonations/tags/)
[![Latest Version](https://img.shields.io/github/release/XoopsModules25x/xdonations.svg?style=flat)](https://github.com/XoopsModules25x/xdonations/releases/)

**xDonations** module for [XOOPS CMS](http://xoops.org), allows a site to accept donations from Paypal and track donations using the Paypal IPN notification.

[![Tutorial Available](http://xoops.org/images/tutorial-available-blue.svg)](https://www.gitbook.com/book/xoops/xdonations-tutorial/) Tutorial: see [here](https://www.gitbook.com/book/xoops/xdonations-tutorial/).
To contribute to the Tutorial, [fork it on GitHub](https://github.com/XoopsDocs/xdonations-tutorial)

[![Translations on Transifex](http://xoops.org/images/translations-transifex-blue.svg)](https://www.transifex.com/xoops)

Please visit us on **http://xoops.org**

The upcoming "next generation" version of XOOPS CMS is being crafted on GitHub at: https://github.com/XOOPS



docs/changelog.txt
2.00 Beta 1       2016-04-05
=====================================================================
- moved all images, CSS, and JS files to /assets (mamba)
- renamed .html Smarty templates to .tpl (mamba)
- removed @version (mamba)
- single quotes (mamba)
- XOOPS 2.5.8 and PHP 7 (mamba)
- PSR-2 adjustments (mamba)



1.97  RC
 =====================================================================
  - updated to ModuleAdmin Class 1.1 (mamba)
  - updated table in "Treasury Register" to standard GUI, with actions on the right side (mamba)

1.97  BETA
=====================================================================
- added new Admin Menu Class (Mage, mamba)
- updated GUI to XOOPS Standard  (mamba)
- updated currencies accepted to match PAYPAL


1.96  FINAL
=====================================================================
  - updated INSTALL.TXT information
  - removed BETA.TXT file

1.96  BETA 2
=====================================================================
  - fixed english language typo in _AD_DON_USERNAME_REQUEST_NO constant
  - fixed 'donators' to 'donors' in _AD_DON_PP_GROUP & _AD_DON_PP_RANK
  - added _AD_DON_T_don_forceadd_ & _AD_DON_V_don_forceadd_ constants
  - fixed typo on line 785 from 'sprint' to sprintf' in /admin/index.php
    causing PHP error
  - refactored code in ipnppd.php to add/remove user from special
    group based on admin config option
  - added option to be able let admin choose whether to add user to
    group/rank even if user selects to remain anonymous
  - removed debug message (dumped sql statements) in
    ./include/installscript.php
  - added eval for install function to allow module to be relocatable
  - New file added - ./include/onupdate.php

1.96  BETA 1
=====================================================================
  - reverted to use XOOPS common.php in XOOPS 2.4.5 instead of using a
    'custom' ./include/common.php - removed file
  - Added additional language constants to improve translation
    capability (Admin help titles still not changed)
  - Added new constant definitions for error tracking
  - Added ability to display/clear IPN log in Admin
  - Added ability to search/display by Transaction ID in Admin
  - Added variable sanitation in various places
  - Added Australian dollar (AUD / AU$) support
  - Added support for xdonations_ipn.log file in XOOPS upload directory
  - Changed 'edit' and 'delete' icons in Admin to use Crystal graphics
  - Changed html code in Admin pages to improve W3C compliance and
    readability
  - Changed date entry in Admin Treasury to use XOOPS javascript calendar
    - removed requirement for FormSolution class (Daniel Convissor)
  - Changed mysql file to use 'ENGINE' instead of 'TYPE'
  - Changed mysql to add table indexes to improve performance
  - Changed mysql queries in various places to reduce number of db calls

- Previous versions
=====================================================================
  - Added language constants to all files to allow for alternate languages
  - Added constant definitions to languages/english/main.php
  - Fixed IPN Debug mode
  - Added include/functions.php for redundant function code and for a bugfix
  - Added success.php and cancel.php for Donation success and cancel pages
  - Added include/installscript.php to set up success and cancel pages in DB
  - Added language include in header.php
  - Changed default monthly donation goals to lower values
  - Changed default "Yes" and "No" options for index.php donation form
  - Added "Left to go" and "Surplus" line to bottom of blocks/Donat-o-meter.php output
  - Changed default block name for blocks/Donat-o-meter.php to "Donations"
  - Changed templates/donations_main.htm to make it look prettier ;-) ("Visual Optimization")
  - Added include/common.php for IPN include (prevents inaccessibility of ipnppd.php in certain cases)
  ... and a mess of other small things.

-- Modification du module pour supporter les traductions, dans le panneau d'administration et dans les javascript.

-- The language traduce is set to the admin panel and the javascript (fait par JJD : jjd@kiolo.com).
-- Correction of javascript bugs ans replacement of "HTTP_GET_VARS"
   and "HTTP_POST_VARS" by "_GET" et "_POST" in "admin/index.php.  (fait par JJD : jjd@kiolo.com)


xoops_version.php
<?php
/************************************************************************/
/* Donations - Paypal financial management module for Xoops 2           */
/* Copyright (c) 2016 XOOPS Project                                     */
/* http://dev.xoops.org/modules/xfmod/project/?group_id=1060            */
/* 
/************************************************************************/
/*                                                                      */
/* Based on NukeTreasury for PHP-Nuke - by Dave Lawrence AKA Thrash     */
/* NukeTreasury - Financial management for PHP-Nuke                     */
/* Copyright (c) 2004 by Dave Lawrence AKA Thrash                       */
/*                       thrash@fragnastika.com                         */
/*                       thrashn8r@hotmail.com                          */
/*                                                                      */
/************************************************************************/
/*                                                                      */
/* This program is free software; you can redistribute it and/or modify */
/* it under the terms of the GNU General Public License as published by */
/* the Free Software Foundation; either version 2 of the License.       */
/*                                                                      */
/* This program is distributed in the hope that it will be useful, but  */
/* WITHOUT ANY WARRANTY; without even the implied warranty of           */
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU     */
/* General Public License for more details.                             */
/*                                                                      */
/* You should have received a copy of the GNU General Public License    */
/* along with this program; if not, write to the Free Software          */
/* Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  */
/* USA                                                                  */
/************************************************************************/

$moduleDirName = basename(__DIR__);

$modversion['name']           = ucfirst($moduleDirName);
$modversion['version']        = '2.00';
$modversion['description']    = _MI_DON_DESC;
$modversion['author']         = 'XOOPS Development Team, zyspec (owners@zyspec.com), Mamba';
$modversion['credits']        = 'Based on NukeTreasury for PHP-Nuke.<br />Xoops2 port maintained by the Xoops2 Donations Module Dev Team.';
$modversion['help']           = 'page=help';
$modversion['license']        = 'GNU GPL 2.0';
$modversion['license_url']    = 'www.gnu.org/licenses/gpl-2.0.html';
$modversion['official']       = 0; //1 indicates supported by XOOPS Dev Team, 0 means 3rd party supported
$modversion['image']          = 'assets/images/logo_module.png';
$modversion['dirname']        = basename(__DIR__);//$xdonDir;
$modversion['dirmoduleadmin'] = '/Frameworks/moduleclasses/moduleadmin';
$modversion['icons16']        = '../../Frameworks/moduleclasses/icons/16';
$modversion['icons32']        = '../../Frameworks/moduleclasses/icons/32';

//about
$modversion['release_file']        = XOOPS_URL . '/modules/' . $modversion['dirname'] . '/docs/changelog.txt';
$modversion['release_date']        = '2016/04/05';
$modversion['module_website_url']  = 'www.xoops.org/';
$modversion['module_website_name'] = 'XOOPS';
$modversion['release']             = '2016-04-05';
$modversion['module_status']       = 'Beta 1';
$modversion['author_website_url']  = 'http://xoops.org/';
$modversion['author_website_name'] = 'XOOPS';
$modversion['min_php']             = '5.5';
$modversion['min_xoops']           = '2.5.8';
$modversion['min_admin']           = '1.1';
$modversion['min_db']              = array('mysql' => '5.0.7', 'mysqli' => '5.0.7');

// Run Script After Installation
$modversion['onInstall'] = 'include/installscript.php';

// Run Script On Update
$modversion['onUpdate'] = 'include/onupdate.php';

// Run Script On Module Removal
$modversion['onUninstall'] = 'include/uninstallscript.php';

// Sql file (must contain sql generated by phpMyAdmin or phpPgAdmin)
// All tables should not have any prefix!
$modversion['sqlfile']['mysql'] = 'sql/mysql.sql';

// Tables created by sql file (without prefix!)
$modversion['tables'][0] = 'donations_config';
$modversion['tables'][1] = 'donations_financial';
$modversion['tables'][2] = 'donations_transactions';
$modversion['tables'][3] = 'donations_translog';

// Admin things
$modversion['hasAdmin']    = 1;
$modversion['adminindex']  = 'admin/index.php';
$modversion['adminmenu']   = 'admin/menu.php';
$modversion['system_menu'] = 1;

// Blocks
$modversion['blocks'][1]['file']        = 'donat_o_meter.php';
$modversion['blocks'][1]['name']        = _MI_DON_BNAME1;
$modversion['blocks'][1]['description'] = _MI_DON_BNAME1_DESC;
$modversion['blocks'][1]['show_func']   = 'b_donations_donatometer_show';
$modversion['blocks'][1]['options']     = '10|1|0|' . _MI_DON_BUTTON_URL . '|110|23';
$modversion['blocks'][1]['edit_func']   = 'b_donations_donatometer_edit';
$modversion['blocks'][1]['template']    = 'donations_donatometer.tpl';

$modversion['blocks'][2]['file']        = 'donate.php';
$modversion['blocks'][2]['name']        = _MI_DON_BNAME2;
$modversion['blocks'][2]['description'] = _MI_DON_BNAME2_DESC;
$modversion['blocks'][2]['show_func']   = 'b_donations_donate_show';
$modversion['blocks'][2]['template']    = 'donations_donate.tpl';

$modversion['blocks'][3]['file']        = 'donors.php';
$modversion['blocks'][3]['name']        = _MI_DON_BNAME3;
$modversion['blocks'][3]['description'] = _MI_DON_BNAME3_DESC;
$modversion['blocks'][3]['show_func']   = 'b_donations_donors_show';
$modversion['blocks'][3]['options']     = '10|1|0';
$modversion['blocks'][3]['edit_func']   = 'b_donations_donors_edit';
$modversion['blocks'][3]['template']    = 'donations_donors.tpl';

// Menu
$modversion['hasMain'] = 1;

global $xoopsUser, $xoopsDB, $xoopsModule, $xoopsModuleConfig;
$hModConfig = xoops_getHandler('config');
$hModule    = xoops_getHandler('module');

if (is_object($xoopsUser) && $xoopsUser->isAdmin()) {
    $modversion['sub'][1]['name'] = _MI_DON_ADMIN;
    $modversion['sub'][1]['url']  = 'admin/index.php';
}

// Templates
$modversion['templates'][1]['file']        = 'donations_main.tpl';
$modversion['templates'][1]['description'] = 'Donations Main Template';
